Стартовая реализация сервиса

KiryakovDenis opened the pull request.

KiryakovDenis committed:
Стартовая реализация сервиса

KiryakovDenis committed:
Добавлены аннотации @Transactional
и версия url

KiryakovDenis committed:
Исправлена обработка ошибок в репозитории

Nokichi commented on src/main/java/ru/kdv/study/tTUser/service/UserService.java:
```diff
@@ -0,0 +1,71 @@
+    private User UserInsertToUser (UserInsert userInsert) {
+        return User.builder()
+                .username(userInsert.getUsername())
+                .passwordHash(userInsert.getPassword().hashCode())
```
Пароль не шифруется. В такой реализации может быть коллизия

KiryakovDenis replied:
В нашем курсе, мы проходим мимо вопросов безопасности, поэтому и в силу требования "не усложнять" выбрал самый простой и самый убогий вариант реализации.

Nokichi commented on src/main/java/ru/kdv/study/tTUser/exception/ExceptionController.java:
```diff
@@ -0,0 +1,35 @@
+import org.springframework.web.bind.annotation.ResponseStatus;
+import ru.kdv.study.tTUser.model.ExceptionMessage;
+
+@ControllerAdvice
```
По-моему нужно использовать @RestControllerAdvice, потому что у нас чистый REST сервис без страниц

Nokichi commented on src/main/java/ru/kdv/study/tTUser/repository/UserRepository.java:
```diff
@@ -0,0 +1,108 @@
+        }
+    }
+
+    private DataBaseException handleException(Exception e) {
```
как будто бы тоже, что и вторая половина метода handleException(Exception e, Long id). Смахивает на дублирование

KiryakovDenis committed:
Доработки по ревью.
Исправил @ControllerAdvice -> @RestControllerAdvice

Избавился от метода handleException

KiryakovDenis committed:
Добавил метаинформацию, перенес заполнение метаинформации на сторону БД

KiryakovDenis committed:
Добавил в БД unique constraint на поле username, обработал это исключение в коде сервиса.

Добавил поля createdAt updatedAt в модель user и добавил их обработку в маппер

KiryakovDenis committed:
Изменить тип кэша пароля в базе.
Изменил принцип формирования кэша
Добавил немного "бантиков" для сваггера

Nokichi commented on src/main/java/ru/kdv/study/tTUser/service/UserService.java:
```diff
@@ -0,0 +1,84 @@
+                .build();
+    }
+
+    private UserResponse UserToResponseUser(User user) {
```
чет 2 метода с большой буквы называются. Обычно по camelCase с маленькой. Я из-за этого читая метод create() выше немного мозг сломал))

KiryakovDenis committed:
Исправил именование функции по ревью

KiryakovDenis committed:
Добавил в ендпоинт выдающий список пользователей параметр массив идентификаторов

gmnpr commented on src/main/java/ru/kdv/study/tTUser/repository/UserRepository.java:
```diff
@@ -0,0 +1,93 @@
+import ru.kdv.study.tTUser.model.User;
+import ru.kdv.study.tTUser.repository.mapper.UserMapper;
+
+import java.time.LocalDateTime;
```
нужный?)

gmnpr commented on src/test/java/ru/kdv/study/tTUser/service/UserServiceTest.java:
```diff
@@ -0,0 +1,88 @@
+package ru.kdv.study.tTUser.service;
```
что с названием пакета?)

gmnpr replied:
ладно, я похоже понял авторскую задумку))

gmnpr commented on src/test/java/ru/kdv/study/tTUser/service/UserServiceTest.java:
```diff
@@ -0,0 +1,88 @@
+    @Test
+    @DisplayName("Валидация имени пользователя с пустым значением")
+    public void validateEmptyPassword() {
+        BadRequestException bre = assertThrows(BadRequestException.class, () -> userService.create(EmptyPasswordUserInsert));
```
зачем сохранять результаты assertThrows?)

gmnpr commented on src/test/java/ru/kdv/study/tTUser/service/UserServiceTest.java:
```diff
@@ -0,0 +1,88 @@
+    private User validUserIn = User.builder()
+            .id(null)
+            .username(validUserInsert.getUsername())
+            .passwordHash(validUserInsert.getPassword().hashCode())
```
тесты не запускаются, проблема тут

KiryakovDenis committed:
Переименовал пакет, Починил тесты, добавил смысла переменной исключения в тестах

KiryakovDenis committed:
переименовал файл свойств

gmnpr commented on src/main/java/ru/kdv/study/taskTrackerUser/security/SecurityUtl.java:
```diff
@@ -0,0 +1,20 @@
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class SecurityUtl {
```
1) Util
2) Если это класс утилита, то у не должно быть публичного конструктора и тд. рекомендую просто накинуть аннотацию UtilityClass

gmnpr commented on src/main/java/ru/kdv/study/taskTrackerUser/repository/UserRepository.java:
```diff
@@ -0,0 +1,92 @@
+        }
+    }
+
+    private MapSqlParameterSource UserToSql(User user) {
```
лучше конечно с маленькой буквы))

KiryakovDenis replied:
Что то со мной не так. Походу нужно к психологу с этой проблемой сходить.

gmnpr commented on src/main/java/ru/kdv/study/taskTrackerUser/repository/UserRepository.java:
```diff
@@ -0,0 +1,92 @@
+
+    public void delete(Long id) {
+        try {
+            User user = jdbcTemplate.queryForObject(DELETE_USER, new MapSqlParameterSource("id", id), userMapper);
```
зачем сохранять user?

gmnpr commented on src/test/java/ru/kdv/study/taskTrackerUser/service/UserServiceTest.java:
```diff
@@ -0,0 +1,100 @@
+    @InjectMocks
+    UserService userService;
+
+
```
лишний /n

gmnpr commented on src/test/java/ru/kdv/study/taskTrackerUser/service/UserServiceTest.java:
```diff
@@ -0,0 +1,100 @@
+        verify(userRepository).insert(validUserIn);
+    }
+
+    private UserInsert NullUsernameUserInsert = new UserInsert(null, "password1");
```
что это между методами и зачем?)

KiryakovDenis committed:
Исправил заглавные буквы в именовании классов и методов. Убрал инициализацию в методе delete в репозитории. Подправил codestyle в тестах

gmnpr commented on src/main/java/ru/kdv/study/taskTrackerUser/repository/UserRepository.java:
```diff
@@ -0,0 +1,92 @@
+
+    public void delete(Long id) {
+        try {
+            jdbcTemplate.queryForObject(DELETE_USER, new MapSqlParameterSource("id", id), userMapper);
```
почти победил! а зачем тут маппер?) мы же не планируем мапить

gmnpr commented on src/test/java/ru/kdv/study/taskTrackerUser/service/UserServiceTest.java:
```diff
@@ -0,0 +1,95 @@
+
+    @InjectMocks
+    UserService userService;
+
```
1 /n надо, давай еще убирать :D

KiryakovDenis committed:
Убрал использование маппера в методе delete, подправил codestyle
